### PR TITLE
adds openapi annotation

### PR DESCRIPTION
**Pull request by invaliduser**

### CODE REVIEW

File: src/main/com/yetanalytics/lrs/pedestal/routes.cljc [hunk 1]
```diff
@@ -41,30 +43,16 @@
...
+       (let [method-kw  (case method
+                          :head :get
+                          method)
+             method-str (-> method-kw name clojure.string/upper-case)
+             token-vec  ["xapi" resource doc-type method-str "request/params"]
+             spec-kw (-> (clojure.string/join \. token-vec)
+                         keyword)
+             doc-params-interceptor 
+             (xapi-i/params-interceptor spec-kw)
+
```
Comment (milt) [inline src/main/com/yetanalytics/lrs/pedestal/routes.cljc:55]: Is this change relevant to adding instrumentation? If it's just that you need the `spec-kw` I'd prefer that you bind it separately using the old method. It was designed the way it was to be explicit/readable with the specs involved (as this app has a LOT of specs)

**Review by milt — CHANGES_REQUESTED** [review]
Only thing is the change to the way doc specs are decided (see my other note). Otherwise looks good.

File: src/main/com/yetanalytics/lrs/pedestal/openapi.cljc [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+               :required [:id]
+               :properties {:objectType {:type :string :pattern "String"}
+                            :id :r#IRI
+                            :definition {:type :object
+                                         :properties {:name {}
+                                                      :description {}
+                                                      :type {}
+                                                      :moreinfo {}
+                                                      :extensions {}}}}}
+    :Agent                              ;maybe important
```
Comment (milt) [inline src/main/com/yetanalytics/lrs/pedestal/openapi.cljc:28]: maybe indeed

File: src/main/com/yetanalytics/lrs/pedestal/openapi.cljc [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  {:health {:operationId :health
+            :responses {200 (gc/response "Empty body---a 200 indicates server is alive")}
+            :description "Simple heartbeat"}
+   :about {:operationId :get-about
+           :description "About info"
+           :responses
+           {200 (gc/response "Object containing body text and optional etag"
+                             (gc/o {:body :t#string
+                                    :#?etag :t#string}))}}
+   :statements-get {:params {:?#statementId :t#string
```
Comment (milt) [inline src/main/com/yetanalytics/lrs/pedestal/openapi.cljc:193]: not-ideal formatting here, maybe do a pass on this ns with clj-pretty-print or the like

File: src/main/com/yetanalytics/lrs/pedestal/routes.cljc (hunk 1 above)
Comment (invaliduser) [reply src/main/com/yetanalytics/lrs/pedestal/routes.cljc]: reverted

File: src/main/com/yetanalytics/lrs/pedestal/openapi.cljc (hunk 2 above)
Comment (invaliduser) [reply src/main/com/yetanalytics/lrs/pedestal/openapi.cljc]: removed

File: src/main/com/yetanalytics/lrs/pedestal/openapi.cljc (hunk 3 above)
Comment (invaliduser) [reply src/main/com/yetanalytics/lrs/pedestal/openapi.cljc]: aligned

**Review by milt — APPROVED** [review]
